driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -84,7 +85,14 @@ public RequestHandler(SessionManager manager, Callback callback, Statement state
 
         callback.register(this);
 
-        this.queryPlan = new QueryPlan(manager.loadBalancingPolicy().newQueryPlan(manager.poolsState.keyspace, statement));
+        // If host is explicitly set on statement, bypass load balancing policy.
+        if (statement.getHost() != null) {
+            this.queryPlan = new QueryPlan(Iterators.singletonIterator(statement.getHost()));
```

Comment:
Should we also disable speculative executions? Or is it too much hassle for little gain?

Reply:
I think that would be a small optimization to bypass scheduling an execution, but I don't think its worth adding additional code.  It looks like an execution could get scheduled, but there is logic in `RequestHandler` to cancel any pending speculative executions if we finish a query or encounter the end of the plan (`reportNoMoreHosts()` is called).  It's possible the execution would get fired, but then encounter nothing in the plan.

Reply:
👍 
System view queries and DDL changes &mdash; the two target use cases for this feature &mdash; are probably not very frequent queries anyway, so that short extra execution should not matter.
And if it does, it can be worked around with a custom speculative execution policy.